### DISCUSSION TITLE
What is the implemented feature set of Simplicity as of today?

### DISCUSSION THREAD

**Discussion by michaelfolkson** [BlockstreamResearch/simplicity#111] [Q&A] [answered]
I guess this is a constantly moving target so hard to track (maybe there should be a doc tracking this?) but what is the current feature set of Simplicity as of today (November 2022)? It currently has feature parity with Script on mainnet, has the [new Tapscript opcodes](https://github.com/ElementsProject/elements/blob/master/doc/tapscript_opcodes.md) on Liquid, has ANYPREVOUT/ANYPREVOUTANYSCRIPT, has "introspection of other input annexes and the scriptsig (used for ctv-like things)" (as discussed in yesterday's meeting on Bitcoin Contracting Primitives yesterday) etc, other things?

**Comment by roconnor-blockstream** [comment, marked as answer]
I don't have an explicit document for this, but as of today you can view the file <https://github.com/ElementsProject/simplicity/blob/master/C/primitive/elements/primitiveEnumJet.inc> which has a list of implemented jets in the `master` branch.  Similarly you can for the `simplicity` branch of elements you can find a list at <https://github.com/ElementsProject/elements/blob/simplicity/src/simplicity/C/primitive/elements/primitiveEnumJet.inc>.

Some of these jets have have a description in Appendix B of the [Simplicity technical report](https://raw.githubusercontent.com/ElementsProject/simplicity/pdf/Simplicity-TR.pdf).  They all have a formal description given as a Simplicity program in various Haskell modules.

That said, I do plan on revamping the generated files eventually, so I expect this information to become outdated at some point in time.  Hopefully at that point in time nearly all the designed jets will have been implemented.

**Reply by michaelfolkson** [reply]
Some discussion on [Twitter](https://twitter.com/niftynei/status/1628408693841620994?s=20) of how the proposed opcode [OP_VAULT](https://github.com/bitcoin/bips/pull/1421) could be implemented in Simplicity and whether it would need a new jet. 

Could be implemented with bare Simplicity and an an existing jet for hashing? ([n1ckler](https://twitter.com/n1ckler/status/1628725313575260161?s=20))
